toolchain/check/handle_array.cpp

```diff
@@ -32,21 +32,25 @@ auto HandleArrayExpr(Context& context, Parse::ArrayExprId parse_node) -> bool {
   context.node_stack()
       .PopAndDiscardSoloParseNode<Parse::NodeKind::ArrayExprSemi>();
   auto element_type_inst_id = context.node_stack().PopExpr();
-  auto bound_inst = context.insts().Get(bound_inst_id);
-  if (auto literal = bound_inst.TryAs<SemIR::IntLiteral>()) {
-    const auto& bound_value = context.ints().Get(literal->int_id);
-    // TODO: Produce an error if the array type is too large.
-    if (bound_value.getActiveBits() <= 64) {
-      context.AddInstAndPush(
-          {parse_node, SemIR::ArrayType{SemIR::TypeId::TypeType, bound_inst_id,
-                                        ExprAsType(context, parse_node,
-                                                   element_type_inst_id)}});
-      return true;
-    }
+
+  // The array bound must be a constant.
+  //
+  // TODO: Should we support runtime-phase bounds in cases such as:
+  //   comptime fn F(n: i32) -> type { return [i32; n]; }
```

Comment:
Not blocking here, but FWIW: I think no? Because it's only valid when the function guaranteed-compile-time, it should be trivial to write it as `template n:! i32`?

If we want something like `if consteval` that might motivate this, but I'm somewhat dubious...